webpack/response_set.js

```diff
@@ -1,27 +1,6 @@
-const $ = require('jquery');
+import React from 'react';
+import ReactDOM from 'react-dom';
+import ResponseSetTable from './components/ResponseSetTable';
 
-exports.addResponseRow = function() {
-  const tableRowCount = $("#response-set-table tr").length - 1;
-  const newRow = `
-        <tr>
-          <td>
-            <label class="hidden" for="response_set_responses_attributes_${tableRowCount}_value">Value</label>
-            <input type="text" name="response_set[responses_attributes][${tableRowCount}][value]" id="response_set_responses_attributes_${tableRowCount}_value">
-          </td>
-          <td>
-            <label class="hidden" for="response_set_responses_attributes_${tableRowCount}_code_system">Code system</label>
-            <input type="text" name="response_set[responses_attributes][${tableRowCount}][code_system]" id="response_set_responses_attributes_${tableRowCount}_code_system">
-          </td>
-          <td>
-            <label class="hidden" for="response_set_responses_attributes_${tableRowCount}_display_name">Display name</label>
-            <input type="text" name="response_set[responses_attributes][${tableRowCount}][display_name]" id="response_set_responses_attributes_${tableRowCount}_display_name">
-          </td>
-      </tr>`;
-  const tbl = $('#response-set-table > tbody:last-child');
-  tbl.append(newRow);
-}
-
-exports.removeResponseRow = function(event) {
-  const row = $(event.target).parents("tr");
-  row.remove();
-}
+const responses = JSON.parse(document.getElementById('responses-json').innerHTML);
```

Comment:
I'm not sure that I completely understand your comment, but one of the reasons this isn't done in the head of the document is that this script is run after the element that contains the JSON data and the element that the React component takes over. I guess another approach would be to do a document onLoad type of thing. My understanding of the latest JavaScript best practices is to not do things like that anymore and push your scripts to the bottom of the page. This lets a good chunk of the rendering happen without the browser needing to download and parse the JS.